### ISSUE TITLE
Port to .Net Core

### ISSUE THREAD

**Issue by sblackler** [dennisdoomen/fluentassertions#265]
As mentioned in gitter, I would like to see FluentAssertions ported to .Net Core. Hopefully it will be a nice and easy port. I'm new to .Net Core myself, but happy to give a hand where I can.

**Comment by smudge202** [comment]
^ This. Going to take a look and see how difficult it would be.

**Comment by smudge202** [comment]
I have one more issue to fix on my fork:

`cannot convert from 'method group' to 'Func<string, Assembly>' in the ProbingAdapterResolver.

The issue is that I don't have a reference to whichever framework assembly dnx451 requires to `Assembly.Load` with a string (only AssemblyName).

I could change the class to work with AssemblyName, but both .Net 4.5 and dnxcore50 support the string overload, so I'm certain I'm just missing a reference.

Don't suppose you know off the top of your head, @kuhlenh?

**Comment by smudge202** [comment]
To keep this up to date (outside of gitter). This issue is currently blocked by aspnet/dnx#1883. The port so far can be found here: https://github.com/smudge202/fluentassertions/tree/CoreCLR2

The feeling I get from the dnx chat rooms is that the issue might never be fixed, or certainly not soon, because mobile apps will from now on be [UWA](https://msdn.microsoft.com/en-us/library/windows/apps/dn609832.aspx). The DNX team _might_ take on mobile support for completion's sake, but it seems more likely that the work is deferred long enough for UWA to take a foothold, making previous mobile platforms obsolete.

Perhaps @davidfowl would be willing to comment on that?

There are two possible outcomes right now (in my opinion):

### DNX will better support Profile32 _soon_

**Solution level port**

As per my current branch listed above. We simply wait for the support to be cooked in and finish off this issue.

### DNX will better support Profile32 _eventually_ or _never_

**Project level port**

Similar to my [original PR](https://github.com/dennisdoomen/fluentassertions/pull/266), add a new `FluentAssertions.DNX` project to the solution and append the build output to the existing NuGet package.


_In both cases, I haven't got the Probing Resolver to work with DNX. Shouldn't be too difficult though._

**Comment by kuhlenh** [comment]
@smudge202 Must have missed this email earlier! I will try to get someone on here who can better answer your Probing Resolver and Profile32 issues.

**Comment by smudge202** [comment]
Awesome, thanks @kuhlenh. If preferred, several of us sit in the FluentAssertions gitter channel.

**Comment by richlander** [comment]
FYI, RE: Profile32 http://embed.plnkr.co/03ck2dCtnJogBKHJ9EjY/preview

**Comment by smudge202** [comment]
@kuhlenh Any luck finding someone that can better help with the Probing Adapter Resolver?

**Comment by kuhlenh** [comment]
@richlander should be helping you soon!

**Comment by richlander** [comment]
I told @davidfowl that he needs to engage on this issue.

**Comment by davidfowl** [comment]
As a workaround, you might want to keep the csproj for profile 32 and instead just point the `project.json` to the output of the csproj for that specific profile.

**Comment by richlander** [comment]
@smudge202 Have you determined which set of projects need to target .NET Core / DNX? It's possible that some do not and that might be an easier set to target.

**Comment by smudge202** [comment]
FluentAssertions.Core and the Shared project need to target .Net Core. With other targets this is done by creating a project for the target referencing both aforementioned projects.

**Comment by richlander** [comment]
It seems like there are two options:

- cross compile.
- the workaround that @davidfowl proposed.

Yes?

**Comment by dennisdoomen** [comment]
Solved in v4.0.
